metrics/handler.go

```diff
@@ -0,0 +1,56 @@
+package metrics
+
+import (
+	"encoding/json"
+	"github.com/rcrowley/go-metrics"
+	"net/http"
+	"path"
+	"strings"
+)
+
+type metricsHandler struct {
+	registry metrics.Registry
+	options  Options
+}
+
+func filterMetrics(reg metrics.Registry, prefix, key string) skipperMetrics {
+	metrics := make(skipperMetrics)
+
+	canonicalKey := strings.TrimPrefix(key, prefix)
+	m := reg.Get(canonicalKey)
+	if m != nil {
+		metrics[key] = m
+	} else {
+		reg.Each(func(name string, i interface{}) {
+			if key == "" || (strings.HasPrefix(name, canonicalKey)) {
+				metrics[prefix+name] = i
+			}
+		})
+	}
+
+	return metrics
+}
+
+func (mh *metricsHandler) sendMetrics(w http.ResponseWriter, p string) {
+	_, k := path.Split(p)
```

Comment:
I guess anyone can be that creative with the metrics prefix. I don't think this would be a problem